File: JavaScript_Courses/JavaScript_Basics/Preparation/02_Conditional_Statements_Exercise/04_Toy_Shop.js
function toyShop(input) {
    
    let puzzleCost = 2.60;
    let dollCost = 3;
    let teddyBearCost = 4.10;
    let minionCost = 8.20;
    let carCost = 2;

    let vacationCost = Number(input[0]);
    let puzzleCount = Number(input[1]);
    let dollCount = Number(input[2]);
    let teddyBearsCount = Number(input[3]);
    let minionsCount = Number(input[4]);
    let carsCount = Number(input[5]);

    let totalCount = puzzleCount + dollCount + teddyBearsCount + minionsCount + carsCount;

    let totalIncome = puzzleCost * puzzleCount + dollCost * dollCount
        + teddyBearCost * teddyBearsCount + minionCost * minionsCount + carCost * carsCount;

    if (totalCount >= 50) {
        totalIncome *= 0.75;

        // let discount = totalIncome * 0.25;
        // totalIncome -= discount;
    }

    totalIncome *= 0.90;

    if (totalIncome >= vacationCost) {
        console.log(`Yes! ${(totalIncome - vacationCost).toFixed(2)} lv left.`);
    } else {
        console.log(`Not enough money! ${(vacationCost - totalIncome).toFixed(2)} lv needed.`);
    }

}

// toyShop(["40.8", "20", "25", "30", "50", "10"]);
// toyShop(["320", "8", "2", "5", "5", "1"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/02_Conditional_Statements_Exercise/04_Toy_Shop.py
vacation_cost = float(input())
puzzles_count = int(input())
talking_dolls_count = int(input())
teddy_bears_count = int(input())
minions_count = int(input())
trucks_count = int(input())

income = (puzzles_count * 2.60 +
          talking_dolls_count * 3 +
          teddy_bears_count * 4.10 +
          minions_count * 8.20 +
          trucks_count * 2)

total_count = puzzles_count + talking_dolls_count + teddy_bears_count + minions_count + trucks_count

discount = 0
if total_count >= 50:
    # income = income * 0.75  # income *= 0.75
    discount = 0.25  # 25% == 25 / 100 == 0.25

total_income = income * (1 - discount)  # income * (1 - 0.25) == income * 0.75

# От спечелените пари Петя трябва да даде 10% за наема на магазина.
final_income = total_income * 0.90

if final_income >= vacation_cost:
    money_left = final_income - vacation_cost
    print(f'Yes! {money_left :.2f} lv left.')
else:
    money_needed = vacation_cost - final_income
    print(f'Not enough money! {money_needed :.2f} lv needed.')
